### ISSUE TITLE
Error: pnp-modern-search result webpart doest not show all the people search results when "Show persona card on hover" is on

### ISSUE THREAD

**Issue by anjuprabhakar** [microsoft-search/pnp-modern-search#3358]
**Version used**
Ex: 4.9/4.8 ( we have tried the latest version as well as 4.8)

**Describe the bug**
We have configured pnp-search and pnp-search result web part for People search in our SharePoint online environment. When "Show persona card on hover" option is disabled we are getting the search result , but hover option is disabled . When "Show persona card on hover" option is enabled we are not getting the results . some results show just icon, details are not showing. 

**To Reproduce**
Configure the search and search results for for people search and then turn on "Show persona card on hover". 

**Expected behavior**
All the search results should be visible with hovering option 

**Screenshots**
![image](https://github.com/microsoft-search/pnp-modern-search/assets/101244033/4974fc6d-9161-46ee-bbc3-fb2bbf134135)

**Desktop (please complete the following information):**
 - Chrome , Safari , Edge

**Comment by kasperbolarsen** [comment]
Please review which permissions the sppkg has been granted in the tenant App Catalog as per https://microsoft-search.github.io/pnp-modern-search/installation/

As I recall it the Graph permissions are required in order to get the values in the hover card.

**Comment by anjuprabhakar** [comment]
> Please review which permissions the sppkg has been granted in the tenant App Catalog as per https://microsoft-search.github.io/pnp-modern-search/installation/
> 
> As I recall it the Graph permissions are required in order to get the values in the hover card.

Graph permissions are approved from office 365 tenant level after uploading the the webpart in the app catalogue.  hover option works for the result which shows the details. but some result are not showing is our problem

**Comment by kasperbolarsen** [comment]
So the results shows up fine, just like this, including picture and all until you enable the hover cards?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/3e3d15f0-95e0-4e18-994d-a943b937608f)

**Comment by anjuprabhakar** [comment]
> So the results shows up fine, just like this, including picture and all until you enable the hover cards? ![image](https://private-user-images.githubusercontent.com/20593570/285861173-3e3d15f0-95e0-4e18-994d-a943b937608f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwOTQzODYsIm5iZiI6MTcwMTA5NDA4NiwicGF0aCI6Ii8yMDU5MzU3MC8yODU4NjExNzMtM2UzZDE1ZjAtOTVlMC00ZTE4LTk5NGQtYTk0M2I5Mzc2MDhmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI3VDE0MDgwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiODViNzgyNzRlODc3YTE1M2M4ODI1MzJjYTQwMmRhMThmY2FiYjRlOTkzNzc1ZTBlMjk2ZDliNWZhZDAyMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g_uNqi22kt8wYDoOSSBIFTmvq4CVwele5CccWXI3MOk)

Yes , Exactly

**Comment by anjuprabhakar** [comment]
> > So the results shows up fine, just like this, including picture and all until you enable the hover cards? ![image](https://private-user-images.githubusercontent.com/20593570/285861173-3e3d15f0-95e0-4e18-994d-a943b937608f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwOTQzODYsIm5iZiI6MTcwMTA5NDA4NiwicGF0aCI6Ii8yMDU5MzU3MC8yODU4NjExNzMtM2UzZDE1ZjAtOTVlMC00ZTE4LTk5NGQtYTk0M2I5Mzc2MDhmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI3VDE0MDgwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiODViNzgyNzRlODc3YTE1M2M4ODI1MzJjYTQwMmRhMThmY2FiYjRlOTkzNzc1ZTBlMjk2ZDliNWZhZDAyMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g_uNqi22kt8wYDoOSSBIFTmvq4CVwele5CccWXI3MOk)
> 
do you have any suggestion for this issue..?

**Comment by anjuprabhakar** [comment]
> So the results shows up fine, just like this, including picture and all until you enable the hover cards? ![image](https://private-user-images.githubusercontent.com/20593570/285861173-3e3d15f0-95e0-4e18-994d-a943b937608f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNzMwMzcsIm5iZiI6MTcwMTI3MjczNywicGF0aCI6Ii8yMDU5MzU3MC8yODU4NjExNzMtM2UzZDE1ZjAtOTVlMC00ZTE4LTk5NGQtYTk0M2I5Mzc2MDhmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI5VDE1NDUzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2YjZmMWI3M2NmZjhmNmQ0NzNhOWVkZDYwNmQzYzBmN2M4MjBjYjBjMTc0ZjcxY2Q0YTdkNzgxZGQyYmU0OGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5cOaQSh9TQSP1d2ooRbWuQiMtg8UWONRAT_JrDOjxPs)

Do  you have any trouble shooting suggestions.  We are not able to resolve this issue yet

**Comment by ivy86** [comment]
I have a comparable issue; however, it does work directly in the browser, but not in the Teams-app. Ive spend many hours on this, but no results. https://github.com/microsoft-search/pnp-modern-search/issues/3278

**Comment by kasperbolarsen** [comment]
Perhaps replacing the pnp people component with the mgt people component could work, as per https://www.m365thinking.com/post/boost-the-person-fields-in-your-pnp-modern-search-layout

**Comment by kasperbolarsen** [comment]
Well well, I just by accident managed to recreate something that sounds like your problem. I installed the package on a brand new tenant and did NOT grant any permissions. I then added a Results web part to a page, made a * search against the localpeopleresults and finally selected the People Layout. Works fine.
Enabled the hover and all the people turned into the base icon
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/2a0c788f-6b47-4455-83b0-5e04c564f1fd)

I then granted the permissions and things are back to normal again.

**Comment by wobba** [comment]
@anjuprabhakar you need to dig into what is different for the accounts where hover is not working. If I understand there are some results/people not coming back? Which search provider are you using? SharePoint or Graph?

**Comment by anjuprabhakar** [comment]
> @anjuprabhakar you need to dig into what is different for the accounts where hover is not working. If I understand there are some results/people not coming back? Which search provider are you using? SharePoint or Graph?

Thanks for the reply
We have test  graph tool kit option on and off ( both case when the Show persona card on hover is on some results are showing and some are not showing. I am checking the difference between the accounts and get back to you shortly

**Comment by anjuprabhakar** [comment]
Thank you for all the support provided.
The issue is resolved.  I checked the difference between the user account is showing correctly and having issue , then found that account information having issue returning location details along with ID, so I checked the layout slot and found person query is mapped to AADObjectID, I chenged the mapping to UserName and the issue got resolved. 
Thanks again for the support @kasperbolarsen @wobba 
Anju
